### PR TITLE
EWS v2 - handling None attachments

### PR DESCRIPTION
**Pull request by roysagi**
## Status
- [x] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/23667

## Description
Ignoring None attachments as they are not legal attachments

### CODE REVIEW

File: Packs/EWS/Integrations/EWSv2/EWSv2.py [hunk 1]
```diff
@@ -1660,12 +1689,14 @@ def get_items_from_folder(folder_path, limit=100, target_mailbox=None, is_public
...
                                              compact_fields=True)
         for attachment in item.attachments:
-            if get_internal_item and isinstance(attachment, ItemAttachment) and isinstance(attachment.item,
-                                                                                           Message):
-                # if found item attachment - switch item to the attchment
-                item_attachment = parse_item_as_dict(attachment.item, account.primary_smtp_address, camel_case=True,
-                                                     compact_fields=True)
-                break
+            if attachment is not None:
+                attachment.parent_item = item
```
Comment (ronykoz) [inline Packs/EWS/Integrations/EWSv2/EWSv2.py:1693]: what do we need this for?

File: Packs/EWS/Integrations/EWSv2/EWSv2.py [hunk 2]
```diff
@@ -1241,22 +1250,40 @@ def get_entry_for_file_attachment(item_id, attachment):
...
-            'attachmentContentId': attachment.content_id,
-            'attachmentContentLocation': attachment.content_location,
-            'attachmentSize': attachment.size,
-            'attachmentLastModifiedTime': attachment.last_modified_time.ewsformat(),
-            'attachmentIsInline': attachment.is_inline,
-            ATTACHMENT_TYPE: FILE_ATTACHMENT_TYPE if isinstance(attachment, FileAttachment) else ITEM_ATTACHMENT_TYPE
-        }
-    except TypeError, e:
+        # if this is a file attachment or a non-empty email attachment
+        if isinstance(attachment, FileAttachment) or hasattr(attachment, 'item'):
```
Comment (ronykoz) [inline Packs/EWS/Integrations/EWSv2/EWSv2.py:1254]: what about `ItemAttachment`

File: Packs/EWS/Integrations/EWSv2/EWSv2.py (hunk 1 above)
Comment (roysagi) [reply Packs/EWS/Integrations/EWSv2/EWSv2.py:1693]: It seems this is just good practice so I left it in: https://github.com/ecederstrand/exchangelib/issues/210
Can take it out if needed

File: Packs/EWS/Integrations/EWSv2/EWSv2.py (hunk 2 above)
Comment (roysagi) [reply Packs/EWS/Integrations/EWSv2/EWSv2.py:1254]: If it is an ItemAttachment we want to make sure it has 'item' attribute - hence it is covered by the second condition.

**Comment by roysagi** [comment]
@amshamah419 - If you could please check the RN.

File: Packs/EWS/Integrations/EWSv2/CHANGELOG.md [hunk 3]
```diff
@@ -1,5 +1,6 @@
 ## [Unreleased]
 - Command - "ews-get-items", add Email entry context by demisto standards as an output.
+- Fixed an issue where an error occurred when getting mails with malformed attachments.
```
Comment (amshamah419) [inline Packs/EWS/Integrations/EWSv2/CHANGELOG.md:3]: Fixed an issue where an error occurred while getting emails with malformed attachments.

Comment (roysagi) [reply Packs/EWS/Integrations/EWSv2/CHANGELOG.md]: Done - please add DocsApproved

**Comment by BigEasyJohn** [comment]
What release did this go out on? I can't seem to find it in release notes.

**Pull request merged**